North-Seattle-College/ad440-winter2022-thursday-repo
# Deploy split sentence/words script to AWS Lambda and expose API endpoints

Closes #118 - Create a Lambda function that splits the paragraph in sentences

### Description
Deployed the local script that splits a paragraph into sentences onto AWS Lambda and exposed API endpoints (via API Gateway) for others to use

Updated `README` in `/backend` to include detailed API documentation

### Testing

**Using Postman is highly recommended**, you can use `curl` but formatting request body is difficult via command line

1. Start a new tab in Postman
2. Change request method to POST
3. Paste in the API endpoint for splitting sentences `https://3s7yrqtdmb.execute-api.us-west-2.amazonaws.com/tokenize/splitSentences`
4. Click on Body -> Change the body type to `raw` and `JSON` 
   - **important: if you do not change to JSON, it's defaulted to text and will throw a status 413 error**
5. Paste in an example request body, you can click to copy the example below
6. Click Send to test the API
```JSON
{
  "text": "Generating random paragraphs can be an excellent way for writers to get their creative flow going at the beginning of the day. The writer has no idea what topic the random paragraph will be about when it appears. This forces the writer to use creativity to complete one of three common writing challenges. The writer can use the paragraph as the first one of a short story and build upon it. A second option is to use the random paragraph somewhere in a short story they create. The third option is to have the random paragraph be the ending paragraph in a short story. No matter which of these challenges is undertaken, the writer is forced to use creativity to incorporate the paragraph into their writing."
}
```

Screenshot:

![image](https://user-images.githubusercontent.com/62415298/158713586-66929c9e-dc0f-42dc-864e-b3394f1bd70c.png)

You can test the other end points by changing the URL to 
- split words: `https://3s7yrqtdmb.execute-api.us-west-2.amazonaws.com/tokenize/splitWords`
- remove stop words: `https://3s7yrqtdmb.execute-api.us-west-2.amazonaws.com/tokenize/removeStopwords`

### Notes

- The Lambda function on AWS is called `sw-test`
- The API Gateway is called `splitSentences2.0`

If you need to change the API or Lambda function, **please make a copy**. Someone edited my original API and it was a pain to debug so I had to make a new one 😬

### Time spent
Roughly 10 hours total
